plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+- `useQueryClient` (via Tanstack Query library) - Lets you coordinate queries and mutations within the same query cache
+
+#### Convenience hooks
+
+- `useCoderQuery` (wrapper over `useQuery`)
+
+## Accessing the SDK from your own custom React components
+
+There are two main ways of accessing the Coder SDK:
+
+- `useCoderSdk`
+- `useCoderQuery`
+
+### Accessing the SDK through `useCoderSdk`
+
+`useCoderSdk` is a lower-level "primitive" for accessing the Coder SDK. The SDK exposes a mix of different REST API methods for interacting with your Coder deployment's resources.
+
+Calling the hook will give you an object with all available API methods. As these methods are all async, **none** of them are suitable for use in render logic. They must be called from within effects or event handlers.
+
+```tsx
+// Illustrative example - these patterns are a very bad idea in production!
+function ExampleComponent() {
+  // The workspace type is exported via the plugin
+  const [workspaces, setWorkspaces] = useState<readonly Workspace[]>([]);
+  const sdk = useCoderSdk();
+
+  // The SDK can be called from any effect
+  useEffect(() => {
+    const getInitialWorkspaces = async () => {
+      const workspacesResponse = await sdk.getWorkspaces({ q: 'owner:me' });
+      const workspaces = workspacesResponse.workspaces;
+      setWorkspaces(workspaces);
+    };
+
+    void getInitialWorkspaces();
+
+    // The SDK maintains a stable memory reference; there is no harm in
+    // including it as part of your dependency arrays. In this case, the
+    // dependency array may as well be empty.
+  }, [sdk]);
+
+  return (
+    <>
+      <button
+        // The SDK can also be called from any event handler
+        onClick={async () => {
+          const newWorkspace = await sdk.createWorkspace(
+            'organizationId',
+            'userId',
+            {
+              // Properties for making new workspace go here
+            },
+          );
+
+          setWorkspaces([...workspaces, newWorkspace]);
+        }}
+      >
+        Create new workspace
+      </button>
+
+      <ul>
+        {workspaces.map(workspace => (
+          <li key={workspace.id}>{workspace.name}</li>
+        ))}
+      </ul>
+    </>
+  );
+}
+```
+
+### The SDK object
+
+The SDK object contains all available API methods. All methods follow the format `<verb>` + `<resource name>`. The SDK has these verbs:
+
+- `get`
+- `post`
+- `put`
+- `patch`
+- `upsert`
+- `delete`
+
+Depending on the Coder resource, there may be different API methods that work with a single resource vs all resources (e.g., `sdk.getWorkspace` vs `sdk.getWorkspaces`).
+
+Note that all of these functions will throw an error if the user is not authenticated.
+
+### Error behavior
+
+All SDK functions will throw in the event of an error. You will need to provide additional error handling to expose errors as values within the UI. (Tanstack Query does this automatically for you.)
```

Comment:
A nit, but I feel like `(Tanstack Query does this automatically for you.)` can be omitted since someone using it probably knows that.  Unless this is a subtle way to try making people switch heh :laughing: